src/crimson/net/ProtocolV1.cc

```diff
@@ -641,6 +637,14 @@ void ProtocolV1::start_accept(SocketFRef&& sock,
           entity_addr_t addr;
           ::decode(addr, p);
           ceph_assert(p.end());
+          if (!(addr.is_legacy() || addr.is_any()) ||
```

Comment:
nit, could you revise the check as following, as the parenthesises and the not operators are difficult to parsed and understand.
```c++
if ((addr.is_legacy() || addr.is_any()) &&
    addr.is_same_host(conn.target_addr)) {
} else {
    // bad!
}
```

Reply:
fixed